### ISSUE TITLE
Provide a way to pause / resume the barcode scanner while leaving the camera preview open

### ISSUE THREAD

**Issue by navaronbracke** [juliansteenbakker/mobile_scanner#1006]
A user requested a way to be able to stop the scanner, while the camera preview is still open and providing video input.

This issue serves as a feature request for that.

When resuming the subscription, the old barcode results should be cleared.

**Comment by burekas7** [comment]
Hi,
@navaronbracke Thanks.
My current implementaion is like that
But yes, it could be nice to do this via controller or something like that.

```
  void startListenToScannerCapture() {
    streamSubscription = controller.barcodes.listen(scannerlistener);
  }

  void scannerlistener(BarcodeCapture barcodeCapture) async {
    Some actions
    .
    .
    .
    .
    await streamSubscription.cancel();
  }

  // And when button pressed
        onPressed: () {
          startListenToScannerCapture();
        },

```

**Comment by burekas7** [comment]
@navaronbracke 

Note:

By the way, there is something weird regarding the `StreamSubscription<BarcodeCapture>`,

`streamSubscriptionScanner = controller.barcodes.listen(scannerlistenerCallback);`

When I call: `streamSubscription.pause()`
And move the scanner on a barcode, it doesn't read it, which is ok.

But when I call `streamSubscription.resume()`,
the stream listener callback is called with the scanned value of the barcode that was scanned while I was on pause.

**Comment by burekas7** [comment]
@navaronbracke 

Hi,
Any update on the issue here?

Thanks.

**Comment by navaronbracke** [comment]
No, not yet. Although there is an open PR, but that is to pause the camera preview, not the barcode scanner.

**Comment by burekas7** [comment]
> No, not yet. Although there is an open PR, but that is to pause the camera preview, not the barcode scanner.

Ok thanks.
But when you are going to work on it, pay attention to my additional comment:
https://github.com/juliansteenbakker/mobile_scanner/issues/1006#issuecomment-2049613983

**Comment by navaronbracke** [comment]
We should clear the scanned barcodes in the scanner itself when resuming. We only do that for the start method currently.

**Comment by AliYar-Khan** [comment]
> [@navaronbracke](https://github.com/navaronbracke)
> 
> Note:
> 
> By the way, regarding option 2 - There is something weird regarding the `StreamSubscription<BarcodeCapture>`,
> 
> `streamSubscriptionScanner = controller.barcodes.listen(scannerlistenerCallback);`
> 
> When I call: `streamSubscription.pause()` And move the scanner on a barcode, it doesn't read it, which is ok.
> 
> But when I call `streamSubscription.resume()`, the stream listener callback is called with the scanned value of the barcode that was scanned while I was on pause.
> 
> version: 5.0.0-beta.2

Happening on version 7.0.0-beta.8 as well

**Comment by AliYar-Khan** [comment]
> We should clear the scanned barcodes in the scanner itself when resuming. We only do that for the start method currently.

had to use 
```
controller.barcodes.drain();
```
To remove existing barcodes. But sometimes it still takes the old values and called the barcode handler function
